### PR TITLE
Support role auth for crr retry

### PR DESCRIPTION
**Pull request by welansari**
Forward port of f2521c6c2eb3e4e9c73d0e4d7459a01798a3ac91

Issue: BB-544

### CODE REVIEW

File: lib/api/BackbeatAPI.js [hunk 1]
```diff
@@ -153,6 +153,10 @@ class BackbeatAPI {
             return errors.InvalidQueryParameter
                 .customizeDescription('must be a non-empty string');
```
Comment (KillianG) [inline lib/api/BackbeatAPI.js]: maybe update the error here to precise `sitename must be a non-empty string` ?
```suggestion
                .customizeDescription('sitename must be a non-empty string');
```

**Review by KillianG — APPROVED** [review]

File: extensions/replication/utils/ObjectFailureEntry.js [hunk 2]
```diff
@@ -34,8 +37,17 @@ class ObjectFailureEntry {
...
+    }
+
     getMember() {
-        return `${this.bucket}:${this.objectKey}:${this.encodedVersionId}`;
+        const member =
+            `${this.bucket}:${this.objectKey}:${this.encodedVersionId}`;
+        if (this.role) {
+            return `${member}:${this.role}`;
+        }
+        return member;
```
Comment (ghost) [inline extensions/replication/utils/ObjectFailureEntry.js:50]:
```suggestion
        if (this.role) {
            return `${member}:${this.role}`;
        }
        return `${this.bucket}:${this.objectKey}:${this.encodedVersionId}`;
```

File: lib/api/BackbeatAPI.js [hunk 3]
```diff
@@ -144,14 +144,18 @@ class BackbeatAPI {
...
         if (marker !== undefined && Number.isNaN(Number.parseInt(marker, 10))) {
             return errors.InvalidQueryParameter
                 .customizeDescription('marker must be a number');
         }
         if (sitename !== undefined && sitename === '') {
             return errors.InvalidQueryParameter
-                .customizeDescription('must be a non-empty string');
+                .customizeDescription('sitename must be a non-empty string');
+        }
+        if (role !== undefined && role === '') {
```
Comment (ghost) [inline lib/api/BackbeatAPI.js]:
```suggestion
        if (role === '') {
```

**Review by ghost — APPROVED** [review]

File: extensions/replication/utils/ObjectFailureEntry.js (hunk 2 above)
Comment (welansari) [reply extensions/replication/utils/ObjectFailureEntry.js:50]: `member` is `${this.bucket}:${this.objectKey}:${this.encodedVersionId}` which is reused when specifying a role.

**Pull request merged**